graylog2-web-interface/src/views/components/visualizations/hooks/useWidgetUnits.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import { useMemo } from 'react';
+import flatten from 'lodash/flatten';
+import compact from 'lodash/compact';
+import uniq from 'lodash/uniq';
+
+import useFieldTypesUnits from 'views/hooks/useFieldTypesUnits';
+import type AggregationWidgetConfig from 'views/logic/aggregationbuilder/AggregationWidgetConfig';
+import type Series from 'views/logic/aggregationbuilder/Series';
+import { parseSeries } from 'views/logic/aggregationbuilder/Series';
+
+const useWidgetUnits = (config: AggregationWidgetConfig) => {
+  const fieldTypesUnits = useFieldTypesUnits();
+  const usedFieldsInSeries = useMemo(() => config.series.map((s: Series) => {
+    const { field } = parseSeries(s.function) ?? {};
+
+    return field;
+  }), [config.series]);
+  const usedFieldsInRowPivots = useMemo(() => flatten(config.rowPivots.map((p) => p.fields)), [config.rowPivots]);
+  const usedFieldsInColumnPivots = useMemo(() => flatten(config.columnPivots.map((p) => p.fields)), [config.columnPivots]);
```

Comment:
We can get away without using the external `flatten` dependency by using `flatMap`:
```suggestion
  const usedFieldsInColumnPivots = useMemo(() => config.columnPivots.flatMap((p) => p.fields), [config.columnPivots]);
```